src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,181 @@
+            head = newNode;
+            size++;
+            return;
+        } else {
+            Node<T> newNode = new Node<>(null, value, null);
+            Node<T> current = node(index);
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev.next = newNode;
+            current.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = node(index);
+        T oldItem = current.item;
+        current.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = node(index);
+        final T remove = current.item;
+        if (index == 0 && size != 1) {
+            head = current.next;
+            current.next.prev = null;
+            current.next = null;
+        } else if (index == size - 1 && size != 1) {
+            tail = current.prev;
+            current.prev.next = null;
+            current.prev = null;
+        } else if (size != 1) {
+            current.prev.next = current.next;
+            current.next.prev = current.prev;
+            current.next = null;
+            current.prev = null;
+        }
+        current.item = null;
+        size--;
+        return remove;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        Node<T> previous = null;
+        if (size == 1 && head.item.equals(object)) {
+            head.item = null;
+            head = null;
+            tail = null;
+            size--;
+            return true;
+        }
+        while (current.next != null) {
+            if (current.item != null && current.item.equals(object)) {
+                if (current == head) {
+                    head = current.next;
+                } else {
+                    previous.next = current.next;
+                }
+                size--;
+                return true;
+            } else if (current.item == null) {
+                if (current == head) {
+                    head = current.next;
+                } else {
+                    previous.next = current.next;
+                }
+                size--;
+                return true;
+            }
+            previous = current;
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> node(int index) {
```

Comment:
```suggestion
    private Node<T> findNode(int index) {
```
краще використовувати дієслова для назв методів

Reply:
ок